plugin/src/main/java/org/opensearch/ml/model/MLModelManager.java

```diff
@@ -999,7 +999,7 @@ public void deployModel(
             listener.onResponse("successful");
             return;
         }
-        if (modelCacheHelper.getLocalDeployedModels().length >= maxModelPerNode) {
+        if (functionName != FunctionName.REMOTE && modelCacheHelper.getLocalDeployedModels().length >= maxModelPerNode) {
```

Comment:
Thanks, have you checked if any other settings have similar issues?

Reply:
There are few task related settings like `plugins.ml_commons.max_register_model_tasks_per_node`, `plugins.ml_commons.max_deploy_model_tasks_per_node`, `"plugins.ml_commons.max_ml_task_per_nod`. But I don't think that should cause any issue.

Reply:
I don't think user will trigger a lot of register and depoy model tasks. But for `plugins.ml_commons.max_ml_task_per_node`, can you check if this limit how many predict tasks can run in parallel on one node ?

Reply:
From the [code](https://github.com/opensearch-project/ml-commons/blob/main/plugin/src/main/java/org/opensearch/ml/task/MLTaskDispatcher.java#L131), it looks like we check this only when dispatch policy is `least_load` , not `round_robin`. By default, task dispatch policy is `round_robin`. I think this should be fine for now.